edgecast/routedns/zone.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2022 Edgecast Inc., Licensed under the terms of the Apache 2.0
+// license. See LICENSE file in project root for terms.
+
+package routedns
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+)
+
+// GetZone retrieves information of the provided ZoneID which includes all dns
+// records, failover servers, and loadbalancing servers if any exists.
+func (svc *RouteDNSService) GetZone(params GetZoneParams,
+) (*ZoneGetOK, error) {
+	apiURL := fmt.Sprintf(
+		"/v2/mcc/customers/%s/dns/zone/%d",
+		params.AccountNumber,
+		params.ZoneID,
+	)
+
+	log.Printf("apiURL:%s", apiURL)
+	request, err := svc.Client.BuildRequest("GET", apiURL, nil)
+
+	if err != nil {
+		return nil, fmt.Errorf("GetZone->Build Request Error: %v", err)
+	}
+
+	parsedResponse := ZoneGetOK{}
+
+	_, err = svc.Client.SendRequest(request, &parsedResponse)
+
+	if err != nil {
+		return nil, fmt.Errorf("GetZone->API Response Error: %v", err)
+	}
+	return &parsedResponse, nil
+}
+
+// AddZone creates a primary zone.
+func (svc *RouteDNSService) AddZone(params AddZoneParams) (*int, error) {
+	apiURL := fmt.Sprintf("/v2/mcc/customers/%s/dns/zone", params.AccountNumber)
+	request, err := svc.Client.BuildRequest("POST", apiURL, params.Zone)
+	if err != nil {
+		return nil, fmt.Errorf(
+			"AddZone->->Build Request Error: %v",
+			err,
+		)
+	}
+
+	resp, err := svc.Client.SendRequestWithStringResponse(request)
+
+	if err != nil {
+		return nil, fmt.Errorf(
+			"AddZone->->API Response Error: %v",
+			err,
+		)
+	}
+
+	// Success response body contains only the zone ID
+	zoneID, err := strconv.Atoi(*resp)
```

Comment:
Same as above regarding printing the string if it failed to parse as an int

Reply:
Confirmed error contains passed string